molgenis-app/src/main/java/org/molgenis/app/promise/mapper/ParelMapper.java

```diff
@@ -280,14 +290,29 @@ private Entity toAgeType(String ageType)
 		List<String> unknown = Lists.newArrayList();
 		for (Entity sample : promiseSampleEntities)
 		{
-			String type = sample.getString("MATERIAL_TYPES");
-			if (materialTypesMap.containsKey(type))
+			String tissue = sample.getString("MATERIAL_TYPES_SUB");
```

Comment:
I mean the other way around, does the code list used for MATERIAL_TYPES_SUB depend on the value for MATERIAL_TYPES?

So, right now, if MATERIAL_TYPES='weefsel', you get MATERIAL_TYPES_SUB=['TISSUE_PARAFFIN_EMBEDDED'|'TISSUE_FROZEN']. And this is the only type that uses MATERIAL_TYPES_SUB.

But if at some point they decide they want to distinguish between blood types as well as tissue types and suddenly if MATERIAL_TYPES='blood', you'll get MATERIAL_TYPES_SUB = ['red'|'white']

Then, without changes, you'll translate 'red' to 'unknown tissue subtype'. But you can already check the MATERIAL_TYPE = 'weefsel' before parsing MATERIAL_TYPES_SUB.

Feel free to laugh if this is not making any sense, I really don't know anything about the mapping, am just triggered by seeing the code parse X_SUB without first checking the value for X.